apps/rush-lib/src/logic/InstallManager.ts

```diff
@@ -737,7 +737,7 @@ export class InstallManager {
 
           // create the new tarball
           tar.create({
-            gzip: true,
+            gzip: false,
             file: tarballFile,
             cwd: tempProjectFolder,
             portable: true,
```

Comment:
Isn't `portable: true` already supposed to ensure the file is portable between OS's?  If it's not working, that seems like a bug in the `tar` library that we could fix directly.

It seems brittle to approach the problem by:  reading a file from disk, and monkeying with its bits, recalculating the checksum, and then overwriting the original input file (which is not an idempotent operation).

Reply:
The tarball is portable across platforms when `portable: true` is set, but the chmod flags aren't exactly the same. PNPM treats this as a hash mismatch - now maybe the fix could be in node-tar or maybe PNPM could ignore chmod-bits in the tar header while it calculates the hash?

Reply:
This is how `tar` reads the file permissions:

[node-tar/lib/pack.js](https://github.com/npm/node-tar/blob/114efef625f3fe82f7afec838b50982867619130/lib/pack.js#L178)
```ts
  [ONSTAT] (job, stat) {
    this.statCache.set(job.absolute, stat)
    job.stat = stat

    // now we have the stat, we can filter it.
    if (!this.filter(job.path, stat))
      job.ignore = true

    this[PROCESS]()
  }
```

The `filter` callback is an option for `tar.create()`, and its unrigorous API design enables the caller to tamper directly with the `stat` object from the cache.  We could exploit that to directly modify the `stat.mode` field.  (It feels a little sloppy, but arguably that is the API contract that they designed heheh.)

Or if you want to be slightly more paranoid, the `statCache` [is also exposed](https://github.com/npm/node-tar/blob/114efef625f3fe82f7afec838b50982867619130/README.md#tarcoptions-filelist-callback-alias-tarcreate) as an API option:

> The following options are mostly internal, but can be modified in some
> advanced use cases, such as re-using caches between runs.
>
> - `linkCache` A Map object containing the device and inode value for
>  any file whose nlink is > 1, to identify hard links.
> - `statCache` A Map object that caches calls `lstat`.

So another idea would be to create a `statCache` with an entry for the `package.json` file before invoking `tar.create()`.

Reply:
Hmm - I like that. If that works then I can avoid having to manually modify the header bytes.

Reply:
Filter in tar.create isn't working for mode. I have pushed the code here if you want to take a look: https://github.com/microsoft/rushstack/tree/sachinjoseph/normalizeChmodInTarHeaderUsingFilter

Reply:
@sachinjoseph `755` is the octal representation.  Whereas this code is specifying a decimal integer.

Reply:
Use `PosixModeBits` to specify the correct number.

Reply:
The actual value that I saw in the debugger is `33206` decimal which is `100666` octal.  This corresponds to:

`100000` = [S_IFREG](https://github.com/nodejs/node/blob/master/doc/api/fs.md#file-type-constants)  regular file
`666` = read+write for user+group+other

If the file type bits are incorrect, then the `tar` library malfunctions.  So you might want to leave the extra bits alone, something like this:
```ts
stat.mode = (stat.mode & ~0x1FF) | PosixModeBits.AllRead;
```

Reply:
Ah, my bad. Let me quickly see if changing it to octal works.

Reply:
@octogonz I have updated the PR

Reply:
And, good point about the file type bit. I did run into when using the filter function but not when I was manually modifying raw the header bytes.